### ISSUE TITLE
When changing a date programmatically, datepicker throws an error!

### ISSUE THREAD

**Issue by mahesh-beehively** [arfedulov/semantic-ui-calendar-react#120]
**Describe the bug**
For `DateInput` when value is changed programmatically (may be based on some other inputs),
specifically when new value(date) is in other than current value's month. And then when trying to open a calendar, It will throw an error as show in screenshot. 
```
Uncaught TypeError: Cannot read property 'position' of undefined
```
**Screenshots**
<img width="1276" alt="Screenshot 2019-03-29 at 5 14 57 PM" src="https://user-images.githubusercontent.com/25790733/55230758-c2267780-5246-11e9-8279-4c5971fee73a.png">

**Dependencies versions**
* semantic-ui-calendar-react 0.14.3

**Comment by arfedulov** [comment]
Thanks for report.

Could you provide more details on how exactly programmatically you trying to change ``value``? I've tried this:

```jsx
<Form.Input onChange={(e, { value }) => this.setState({ date: value })} />
<DateInput onChange={(e, { value }) => this.setState({ date: value })} />
```
Then I click on date input and see that value in calendar didn't change (even though value inside input field changed). But I don't see errors.

**Comment by mahesh-beehively** [comment]
@arfedulov Thanks for reply. 
After spending sometime with this I found that this happens only when I've set `minDate` option!
Basically, what I'm trying to do is, I need second date always to be equal or greater than first one. (I know dateRange can help here but that's not fitting in my UI)
Here is the li'll demo to reproduce this.

```
import React from 'react';
import { DateInput } from 'semantic-ui-calendar-react';
import moment from 'moment';
import { Form  } from 'semantic-ui-react';

export default class DateDemo extends React.Component {

  state = {
    startDate: moment().format("MMM DD, YYYY"),
    endDate: moment().add(1, 'day').format("MMM DD, YYYY")
  }

  handleChange = (event, {name, value}) => {
    this.setState({
      [name]: value
    })
    if (name === "startDate") { //&& endDate > startDate
      this.setState({
        endDate: value
      })
    }
  }

  render() {
    const {startDate, endDate} = this.state;
    return (
      <Form>
        <Form.Field>
        <DateInput
           closable
           inlineLabel
           closeOnMouseLeave={false}
           initialDate={startDate}
           pickerWidth={"250px"}
           name="startDate"
           dateFormat={"MMM DD, YYYY"}
           placeholder="Start Date"
           value={startDate}
           onChange={this.handleChange}
        />
        </Form.Field>
        <Form.Field>
        <DateInput
          closable
          inlineLabel
          closeOnMouseLeave={false}
          minDate={startDate}
          initialDate={endDate}
          pickerWidth={"250px"}
          name="endDate"
          dateFormat={"MMM DD, YYYY"}
          placeholder="End Date"
          value={endDate}
          onChange={this.handleChange}
        />
        </Form.Field>
      </Form>
    )
  }
}

```

**Comment by arfedulov** [comment]
Yeah, I got an error. When you select any date from any month other than initial month on first DateInput,
the second DateInput doesn't reflect this change in it's internal current date. So when the second DateInput is trying to parse its input value, the error  occurs. 

This is a bug. DateInput should handle this case when parsing input value.

As a quick fix you can set ``key={endDate}`` on the second DateInput. So it will always be reinstantiated when ``endDate`` changes.

**Comment by mahesh-beehively** [comment]
Cool! Thanks for quick turn-around and providing a quick fix.

**Comment by monofone** [comment]
For me setting the initialDate of the second Input to the minDate solved the problem for now

**Comment by arfedulov** [comment]
Fixed in #142
